components/FilterComponent.vue

```diff
@@ -0,0 +1,171 @@
+        <b-button @click="activateFilter" variant="success" class="mt-2" :disabled="allowActivate != ''">
+          Activate filter
+        </b-button>
+      </div>
+    </div>
+    <div v-else>
+      <b-row>
+        <b-col cols="9" class="mt-auto">
+          <h5>{{ filterDescription }}</h5>
+        </b-col>
+        <b-col cols="3">
+          <b-button @click="deleteFilter" variant="danger" class="mt-2">Delete filter</b-button>
+        </b-col>
+      </b-row>
+    </div>
+    <hr>
+  </div>
+</template>
+
+<script lang="ts">
+import {Component, Prop, Vue}                          from 'nuxt-property-decorator';
+import {SplitFile, selectTime}                         from '~/util/splits';
+import {stringTimeToSeconds, secondsToFormattedString} from '~/util/durations';
+import {Filter}                                        from '~/util/filter';
+import store                                           from '~/util/store';
+import Multiselect                                     from 'vue-multiselect';
+
+@Component({components: {Multiselect}})
+export default class FilterComponent extends Vue {
+  @Prop()
+  filterIndex!: number;
+
+  @Prop()
+  parsedSplits!: SplitFile;
+
+  useRealTime: boolean = store.state.useRealTime;
+
+  globalFilters: Filter[] = store.state.filters;
+
+  filterData: Filter = {details: {label: "", index: -2}, timeMin: 0, timeMax: 0, active: false, attempts: []};
+
+  get filterLabels() {
+    let options = [{index: -1, label: "Global"}];
+    this.parsedSplits.Run.Segments.Segment.forEach((split, index) => options.push({index: index, label: split.Name}));
+
+    return options;
+  }
+
+  get allowActivate() {
+    if (this.filterData.details?.index == -2 || this.filterData.details?.index == undefined)
+      return "You need to select a split.";
+
+    if (this.filterData.timeMin == this.filterData.timeMax)
+      return "You need to select a time range.";
+
+    return "";
+  }
+
+  get filterList() {
+    const index = this.filterData.details?.index;
+    if (index != undefined) {
+      if (index == -1) {
+        return this.filterListGlobal();
+      } else if (index > -1) {
+        return this.filterListSplits(index);
+      }
+    }
+
+    return [];
+  }
+
+  filterListGlobal() {
+    let filteredAttempts: number[] = [];
+    const attempts = this.parsedSplits.Run.AttemptHistory.Attempt;
+    const timeMin = this.filterData.timeMin;
+    const timeMax = this.filterData.timeMax;
+
+    attempts.forEach(attempt => {
+      const attemptTime = selectTime(attempt);
+
+      if (attemptTime != undefined && timeMin != undefined && timeMax != undefined) {
+        const secondsAttemptTime = stringTimeToSeconds(attemptTime);
+
+        if (secondsAttemptTime > timeMin && secondsAttemptTime < timeMax && attempt['@_id'] > 0) {
+          filteredAttempts.push(attempt['@_id']);
+        }
+      }
+    });
+
+    return filteredAttempts;
+  }
+
+  filterListSplits(index: number) {
```

Comment:
This looks a lot like `filterListGlobal`, maybe there is a way to pass an `OptionalRealAndGameTime` array to a unique function that works for both